google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcResultSetMetaData.java

```diff
@@ -0,0 +1,205 @@
+    this.statement = statement;
+  }
+
+  @Override
+  public boolean isClosed() {
+    return false;
+  }
+
+  @Override
+  public int getColumnCount() throws SQLException {
+    return spannerResultSet.getColumnCount();
+  }
+
+  @Override
+  public boolean isAutoIncrement(int column) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public boolean isCaseSensitive(int column) throws SQLException {
+    int type = getColumnType(column);
+    return type == Types.NVARCHAR || type == Types.BINARY;
+  }
+
+  @Override
+  public boolean isSearchable(int column) throws SQLException {
+    return true;
+  }
+
+  @Override
+  public boolean isCurrency(int column) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public int isNullable(int column) throws SQLException {
+    return columnNullableUnknown;
+  }
+
+  @Override
+  public boolean isSigned(int column) throws SQLException {
+    int type = getColumnType(column);
+    return type == Types.DOUBLE || type == Types.BIGINT;
+  }
+
+  @Override
+  public int getColumnDisplaySize(int column) throws SQLException {
+    int colType = getColumnType(column);
+    switch (colType) {
+      case Types.ARRAY:
+        return 50;
+      case Types.BOOLEAN:
+        return 5;
+      case Types.BINARY:
+        int binaryLength = getPrecision(column);
+        return binaryLength == 0 ? 50 : binaryLength;
+      case Types.DATE:
+        return 10;
+      case Types.DOUBLE:
+        return 14;
+      case Types.BIGINT:
+        return 10;
+      case Types.NVARCHAR:
+        int length = getPrecision(column);
+        return length == 0 ? 50 : length;
+      case Types.TIMESTAMP:
+        return 16;
+      default:
+        return 10;
+    }
+  }
+
+  @Override
+  public String getColumnLabel(int column) throws SQLException {
+    return spannerResultSet.getType().getStructFields().get(column - 1).getName();
+  }
+
+  @Override
+  public String getColumnName(int column) throws SQLException {
+    return spannerResultSet.getType().getStructFields().get(column - 1).getName();
+  }
+
+  @Override
+  public String getSchemaName(int column) throws SQLException {
+    return statement.getConnection().getSchema();
+  }
+
+  @Override
+  public int getPrecision(int column) throws SQLException {
+    int colType = getColumnType(column);
+    switch (colType) {
+      case Types.BOOLEAN:
+        return 1;
+      case Types.DATE:
+        return 10;
+      case Types.DOUBLE:
+        return 14;
+      case Types.BIGINT:
+        return 10;
+      case Types.TIMESTAMP:
+        return 24;
+      default:
+        // Not fixed size, we could try to get it from INFORMATION_SCHEMA
+        return 50;
+    }
+  }
+
+  @Override
+  public int getScale(int column) throws SQLException {
+    int colType = getColumnType(column);
+    if (colType == Types.DOUBLE)
+      return 15;
+    return 0;
+  }
+
+  @Override
+  public String getTableName(int column) throws SQLException {
+    return "";
```

Comment:
why empty string?

Reply:
Cloud Spanner does not return the table name in its metadata. We could try to figure it out by parsing the SQL statement that was used to get the result set, but I don't think that's something we want to do for the first version. The javadoc of this method on the interface states `@return table name or "" if not applicable`.